test/compiler.test.js

```diff
@@ -1016,41 +1016,106 @@ describe('compiler', function() {
     });
 
     describe('for mixins', function() {
-      function verifyMixinIsFoundViaMixinDirs(sourceFile, mixinDirs) {
-        var appJS = appdir.writeFileSync(sourceFile, '');
+      describe(' - mixinDirs', function() {
+        function verifyMixinIsFoundViaMixinDirs(sourceFile, mixinDirs) {
+          var appJS = appdir.writeFileSync(sourceFile, '');
 
-        var instructions = boot.compile({
-          appRootDir: appdir.PATH,
-          mixinDirs: mixinDirs
+          var instructions = boot.compile({
+            appRootDir: appdir.PATH,
+            mixinDirs: mixinDirs
+          });
+
+          expect(instructions.mixins[0].sourceFile).to.eql(appJS);
+        }
+
+        it('supports `mixinDirs` option', function() {
+          verifyMixinIsFoundViaMixinDirs('mixins/other.js', ['./mixins']);
         });
 
-        expect(instructions.mixins[0].sourceFile).to.eql(appJS);
-      }
+        it('resolves relative path in `mixinDirs` option', function() {
+          verifyMixinIsFoundViaMixinDirs('custom-mixins/other.js',
+            ['./custom-mixins']);
+        });
 
-      it('supports `mixinDirs` option', function() {
-        verifyMixinIsFoundViaMixinDirs('mixins/other.js', ['./mixins']);
+        it('resolves module relative path in `mixinDirs` option', function() {
+          verifyMixinIsFoundViaMixinDirs('node_modules/custom-mixins/other.js',
+            ['custom-mixins']);
+        });
       });
 
-      it('resolves relative path in `mixinDirs` option', function() {
-        verifyMixinIsFoundViaMixinDirs('custom-mixins/vehicle.js',
-          ['./custom-mixins']);
-      });
+      describe(' - mixinSources', function() {
+        function verifyMixinIsFoundViaMixinSources(sourceFile, mixinSources) {
+          var appJS = appdir.writeFileSync(sourceFile, '');
+
+          var instructions = boot.compile({
+            appRootDir: appdir.PATH,
+            mixinSources: mixinSources
+          });
+
+          expect(instructions.mixins[0].sourceFile).to.eql(appJS);
```

Comment:
If my understanding is correct, then this function should always fail if the compiler implementation works correctly. The mixin instruction should be included only when there is a model using the mixin. Since there are no models defined here, the compiler should not load any mixins from `mixinSources`.

Am I missing something?